fhir-path/src/main/java/com/ibm/fhir/path/function/MemberOfFunction.java

```diff
@@ -119,16 +124,108 @@ public int getMaxArity() {
                     }
                 }
                 return membershipCheckFailed(evaluationContext, elementNode, url, strength);
-            } else {
+            }
+            else if (isSyntaxBased(valueSet)) {
+                
+                // Validate against syntax-based value set
+                if (validateAgainstSyntaxBasedValueSet(valueSet, evaluationContext, elementNode, strength)) {
+                    return SINGLETON_TRUE;
+                }
+                return membershipCheckFailed(evaluationContext, elementNode, url, strength);
+            }
+            else {
                 generateIssue(evaluationContext, IssueSeverity.WARNING, IssueType.INCOMPLETE, "Membership check was not performed: value set '" + url + "' is empty or could not be expanded", elementNode.path());
             }
-        } else {
+        }
+        else {
             generateIssue(evaluationContext, IssueSeverity.WARNING, IssueType.NOT_SUPPORTED, "Membership check was not performed: value set '" + url + "' is not supported", elementNode.path());
         }
 
         return SINGLETON_TRUE;
     }
 
+    /**
+     * Determines if the value set is syntax-based.
+     * @param valueSet the value set
+     * @return true or false
+     */
+    private boolean isSyntaxBased(ValueSet valueSet) {
+        String valueSetUrl = valueSet.getUrl() != null ? valueSet.getUrl().getValue().split("\\|")[0] : null;
```

Comment:
what is the split for?

Reply:
In case the profile specified a canonical with a version, I used the same split as done in CodeGenerator to just compare the URL part.